src/interface.jl

```diff
@@ -0,0 +1,655 @@
+    end
+
+    return lp
+end
+
+#######################################################
+# Constrained to unconstrained distribution bijectors #
+#######################################################
+"""
+    DistributionBijector(d::Distribution)
+    DistributionBijector{<: ADBackend, D}(d::Distribution)
+
+This is the default `Bijector` for a distribution. 
+
+It uses `link` and `invlink` to compute the transformations, and `AD` to compute
+the `jacobian` and `logabsdetjac`.
+"""
+struct DistributionBijector{AD, D} <: ADBijector{AD} where D <: Distribution
+    dist::D
+end
+function DistributionBijector(dist::D) where D <: Distribution
+    DistributionBijector{ADBackend(), D}(dist)
+end
+
+# Simply uses `link` and `invlink` as transforms with AD to get jacobian
+(b::DistributionBijector)(x) = link(b.dist, x)
+(ib::Inversed{<: DistributionBijector})(y) = invlink(ib.orig.dist, y)
+
+
+"Returns the constrained-to-unconstrained bijector for distribution `d`."
+bijector(d::Distribution) = DistributionBijector(d)
+
+# Transformed distributions
+struct TransformedDistribution{D, B, V} <: Distribution{V, Continuous} where {D <: Distribution{V, Continuous}, B <: Bijector}
+    dist::D
+    transform::B
+end
+function TransformedDistribution(d::D, b::B) where {V <: VariateForm, B <: Bijector, D <: Distribution{V, Continuous}}
+    return TransformedDistribution{D, B, V}(d, b)
+end
+
+
+const UnivariateTransformed = TransformedDistribution{<: Distribution, <: Bijector, Univariate}
+const MultivariateTransformed = TransformedDistribution{<: Distribution, <: Bijector, Multivariate}
+const MvTransformed = MultivariateTransformed
+const MatrixTransformed = TransformedDistribution{<: Distribution, <: Bijector, Matrixvariate}
+const Transformed = TransformedDistribution
+
+
+"""
+    transformed(d::Distribution)
+    transformed(d::Distribution, b::Bijector)
+
+Couples distribution `d` with the bijector `b` by returning a `TransformedDistribution`.
+
+If no bijector is provided, i.e. `transformed(d)` is called, then 
+`transformed(d, bijector(d))` is returned.
+"""
+transformed(d::Distribution, b::Bijector) = TransformedDistribution(d, b)
+transformed(d) = transformed(d, bijector(d))
+
+"""
+    bijector(d::Distribution)
+
+Returns the constrained-to-unconstrained bijector for distribution `d`.
+"""
+bijector(d::Normal) = IdentityBijector
+bijector(d::MvNormal) = IdentityBijector
+bijector(d::PositiveDistribution) = log_b
+bijector(d::SimplexDistribution) = simplex_b_proj
+
+_union2tuple(T1::Type, T2::Type) = (T1, T2)
+_union2tuple(T1::Type, T2::Union) = (T1, _union2tuple(T2.a, T2.b)...)
+_union2tuple(T::Union) = _union2tuple(T.a, T.b)
+
+bijector(d::Kolmogorov) = Logit(zero(eltype(d)), zero(eltype(d)))
+for D in _union2tuple(UnitDistribution)[2:end]
+    # Skipping Kolmogorov because it's a DataType
+    @eval bijector(d::$D{T}) where T <: Real = Logit(zero(T), one(T))
+end
+
+# FIXME: Can we make this type-stable?
```

Comment:
Better to leave a name here so people working on it later would know how to contact.

Reply:
Good idea; will do!